autofs-5.1.2 - fix bogus check in expire_cleanup()

From: Ian Kent <raven@themaw.net>

ap->submount can never be 0 here.

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG      |    1 +
 daemon/state.c |    4 +---
 2 files changed, 2 insertions(+), 3 deletions(-)

--- autofs-5.0.7.orig/CHANGELOG
+++ autofs-5.0.7/CHANGELOG
@@ -233,6 +233,7 @@
 - capture cache option and its settings during parsing.
 - handle map_option cache for top level mounts.
 - handle amd cache option all in amd type auto mounts.
+- fix bogus check in expire_cleanup().
 
 25/07/2012 autofs-5.0.7
 =======================
--- autofs-5.0.7.orig/daemon/state.c
+++ autofs-5.0.7/daemon/state.c
@@ -139,9 +139,7 @@ void expire_cleanup(void *arg)
 					next = ST_SHUTDOWN_PENDING;
 					break;
 				}
-
-				if (ap->submount++ == 0)
-					ap->submount = 2;
+				ap->submount++;
 			}
 
 			if (ap->state == ST_EXPIRE && !ap->submount)
